content/posts/unpublished/backbase-tech-stack/post.md

```diff
@@ -0,0 +1,198 @@
+# Backbase tech stack
+
+What exactly is being used to build banking applications in Backbase?
+
+![](assets/tech_stack_img_2.jpeg)
+
+Authors: Paweł Garbacik
+Date: unpublished
+Category: tech-life
+
+tags: architecture,technology,patterns,stack
+
+---
+
+## Introduction
+
+Backbase Engagement Banking Services is a comprehensive solution with a lot of capabilities.
+It is thought out and designed to provide modern architecture with the latest market standards,
+the most advanced security mechanisms, and great extensibility and flexibility to make it easy
+to integrate with different core banking systems.
+
+To achieve this, different technologies, frameworks, standards, and architectural patterns need to be combined.
+
+
+This article aims to provide an insight into how the Backbase software is built and integrated on a high level,
+what kind of technologies are used on a different levels and layers.
+
+## High level architecture
+
+<img align="right" src="assets/HL_Architecture.drawio.png">
+
+Engagement Banking Services are divided according to the capabilities they offer, the role they play, and on which level they operate.
+
+Each service is designed to perform completely different tasks and offers specific ways to interact with it.
+
+On the top there are client-facing apps. These applications are available for 2 different platforms:
+
+* Web applications - available on modern web-browsers, dedicated for different group of users, segregated by the roles and responsibilities
+* Mobile applications - available on popular mobile devices powered by iOS and Android Mobile OSs
+
+These applications allow to interact with backend services for these users who have proper permissions.
+Client APIs are a dedicated and well secured APIs designed for interaction between client-facing applications and backend services.
+
```

Comment:
if you are talking about the api layers would be interesting to document what types of security each one have:
client-api - OIDC Authorization flow with PKCE
service-api - OAuth2 Client Credentials
integration-api - mTLS

Reply:
These security info have been added for each type of api